iree/compiler/Codegen/LLVMCPU/LLVMCPUTileFuseAndVectorizeLinalgTensorOps.cpp

```diff
@@ -28,6 +28,15 @@
 namespace mlir {
 namespace iree_compiler {
 
+// A flag to switch between inline asm and intrinsics while we develop these two
+//  parallel paths.
+static llvm::cl::opt<bool> clUseMmt4dUseIntrinsics(
```

Comment:
If this is a developer flag for experimentation, please add a big scary NOTE to say that it is and shouldnt be relied on. If you need something more long term here, what would that be? I am assuming this flag is short lived.

Reply:
It's for development and for our automated builds. Definitely not intended to be user-facing. I can expand the comment here, but I put "Not for production use" in the description, which seemed more visible. I'm not sure how short-term it is, necessarily, since it would go away once we decide that we can make intrinsics always better than asm and can delete the asm path. If we make that true for some configuration it wouldn't be surprising if it became not true for another.

I was talking to Benoit about his comment that we really might have multiple paths and various combinations. Rather than the very complicated thing I suggested, I would propose that we have a single enum (or bool for now) flag, that says "prefer x" and then we just add x as priority if it applies and fall back to default otherwise. Then we can test all the different configurations in our builds (automated and manual) without creating a really complicated API. So for now (until we have more than two), I would change this to be `--iree-codegen-mmt4d-prefer-intrinsics`. WDYT?

Reply:
Yeah that makes sense. I am fine with flags, etc. for experimental work to help with workflow. They are all just developer hooks (just like LLVM has a ton of flags, none of which are exposed through clang). So fine with this approach.